encrypt_util.go

```diff
@@ -190,46 +190,56 @@ func encryptFile(
 	return meta, tmpOutputFile.Name(), nil
 }
 
-func decryptFile(
+func decryptMetadata(
 	metadata *encryptMetadata,
 	sfe *snowflakeFileEncryption,
-	filename string,
-	chunkSize int,
-	tmpDir string) (
-	string, error) {
+	chunkSize int) (int, cipher.BlockMode, error) {
 	if chunkSize == 0 {
 		chunkSize = aes.BlockSize * 4 * 1024
 	}
 	decodedKey, err := base64.StdEncoding.DecodeString(sfe.QueryStageMasterKey)
 	if err != nil {
-		return "", err
+		return 0, nil, err
 	}
 	keyBytes, err := base64.StdEncoding.DecodeString(metadata.key) // encrypted file key
 	if err != nil {
-		return "", err
+		return 0, nil, err
 	}
 	ivBytes, err := base64.StdEncoding.DecodeString(metadata.iv)
 	if err != nil {
-		return "", err
+		return 0, nil, err
 	}
 
 	// decrypt file key
 	decryptedKey := make([]byte, len(keyBytes))
 	if err = decryptECB(decryptedKey, keyBytes, decodedKey); err != nil {
-		return "", err
+		return 0, nil, err
 	}
 	decryptedKey, err = paddingTrim(decryptedKey)
 	if err != nil {
-		return "", err
+		return 0, nil, err
 	}
 
 	// decrypt file
 	block, err := aes.NewCipher(decryptedKey)
 	if err != nil {
-		return "", err
+		return 0, nil, err
 	}
 	mode := cipher.NewCBCDecrypter(block, ivBytes)
 
+	return chunkSize, mode, err
+}
+
+func decryptFile(
+	metadata *encryptMetadata,
+	sfe *snowflakeFileEncryption,
+	filename string,
+	chunkSize int,
+	tmpDir string) (string, error) {
+	chunkSize, mode, err := decryptMetadata(metadata, sfe, chunkSize)
+	if err != nil {
+		return "", err
+	}
 	tmpOutputFile, err := os.CreateTemp(tmpDir, baseName(filename)+"#")
 	if err != nil {
 		return "", err
```

Comment:
Instead of reading file in chunks, maybe we can just open a stream and leverage `decryptStream`?

Reply:
Refactoring `decryptFile()` seems irrelevant to this PR. I extracted the common code to `decryptFileKey()` and kept the rest of the logic the same as before.